File: tests/syntax/supported/swift/contract.swift
/// Contract fixture exercising a broad slice of Swift syntax.
import Foundation

let MAX_COUNT = 100
let pi: Double = 3.14159

enum Direction {
    case north
    case south
    case east
    case west
}

protocol Shape {
    var area: Double { get }
    func describe() -> String
}

struct Circle: Shape {
    let radius: Double

    var area: Double {
        return pi * radius * radius
    }

    func describe() -> String {
        return "Circle(r=\(radius))"
    }
}

class Counter {
    private var value: Int = 0

    func increment(by amount: Int = 1) {
        value += amount
    }

    var current: Int {
        return value
    }
}

enum ParseError: Error {
    case overflow
    case invalid(Character)
}

func parseDigit(_ c: Character) throws -> Int {
    guard let digit = c.wholeNumberValue else {
        throw ParseError.invalid(c)
    }
    return digit
}

func classify(_ dir: Direction) -> String {
    switch dir {
    case .north, .south:
        return "vertical"
    default:
        return "horizontal"
    }
}

func sum(_ values: [Int]) -> Int {
    var total = 0
    for value in values {
        total += value
    }
    return total
}

extension Circle {
    static func unit() -> Circle {
        return Circle(radius: 1.0)
    }
}

func main() async {
    let shapes: [Shape] = [Circle(radius: 2.0)]
    let numbers = [1, 2, 3, 4, 5]
    let doubled = numbers.map { $0 * 2 }

    let counter = Counter()
    counter.increment(by: 5)

    let digit = try? parseDigit("7")
    if let d = digit {
        print("digit=\(d) total=\(sum(doubled)) area=\(shapes[0].area)")
    }
}

File: tests/syntax/supported/zig/contract.zig
//! Contract fixture exercising a broad slice of Zig syntax.
const std = @import("std");
const builtin = @import("builtin");

const MAX_ENTRIES: usize = 128;
const PI: f64 = 3.14159;

const Direction = enum {
    north,
    south,
    east,
    west,
};

const Vec2 = struct {
    x: f32 = 0.0,
    y: f32 = 0.0,

    pub fn length(self: Vec2) f32 {
        return @sqrt(self.x * self.x + self.y * self.y);
    }

    pub fn add(self: Vec2, other: Vec2) Vec2 {
        return Vec2{ .x = self.x + other.x, .y = self.y + other.y };
    }
};

const ParseError = error{
    Overflow,
    InvalidCharacter,
};

fn parseDigit(c: u8) ParseError!u8 {
    if (c < '0' or c > '9') {
        return ParseError.InvalidCharacter;
    }
    return c - '0';
}

pub fn sumSlice(values: []const i32) i64 {
    var total: i64 = 0;
    for (values) |value| {
        total += value;
    }
    return total;
}

pub fn classify(dir: Direction) []const u8 {
    return switch (dir) {
        .north => "n",
        .south => "s",
        else => "?",
    };
}

pub fn main() !void {
    var origin = Vec2{};
    const shifted = origin.add(Vec2{ .x = 3.0, .y = 4.0 });
    const dist = shifted.length();

    const numbers = [_]i32{ 1, 2, 3 };
    const total = sumSlice(&numbers);

    const digit = parseDigit('7') catch 0;
    std.debug.print("d={d} dist={d} total={d}\n", .{ digit, dist, total });

    comptime {
        const flag = builtin.is_test;
        _ = flag;
    }
}
